### PR TITLE
SONARJAVA-1070 Migrates too many statements per line

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java [hunk 1]
```diff
@@ -47,91 +74,231 @@
...
+      } else {
+        addLine(endToken);
       }
-      if (lineStart != lineEnd) {
-        if (!isNestedInStatement(lineEnd, columnEnd)) {
-          statementsPerLine.add(lineEnd);
+    }
+
+    @CheckForNull
+    private SyntaxToken getFirstToken(Tree expression) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java]: :new: MAJORThe Cyclomatic Complexity of this method "getFirstToken" is 21 which is greater than 10 authorized.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
:new: MAJORReduce the number of returns of this method 11, down to the maximum allowed 5.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1142)

File: java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java [hunk 2]
```diff
@@ -47,91 +74,231 @@
...
+        }
+      }
+    }
   }
 
-  @Override
-  public void visitNode(AstNode statementNode) {
-    if (statementNode.is(JavaLexer.VARIABLE_DECLARATORS) && !statementNode.hasParent(JavaLexer.BLOCK_STATEMENTS)) {
-      return;
+  private class StatementVisitor extends BaseTreeVisitor {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java]: :new: MAJORShould org.sonar.java.checks.TooManyStatementsPerLine_S00122_Check$StatementVisitor be a _static_ inner class?[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASIC_INNER_SHOULD_BE_STATIC)

File: java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java [hunk 3]
```diff
@@ -47,91 +74,231 @@
...
-      Kind.EXPRESSION_STATEMENT,
-      Kind.FOR_STATEMENT,
-      Kind.FOR_EACH_STATEMENT,
-      Kind.TRY_STATEMENT)
-    .build()
-    .toArray(new AstNodeType[0]);
-
-  private final Multiset<Integer> statementsPerLine = HashMultiset.create();
-  private final Map<Integer, Integer> columnsByLine = new HashMap<Integer, Integer>();
+public class TooManyStatementsPerLine_S00122_Check extends SubscriptionBaseVisitor {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java]: MAJORRename this class name to match the regular expression '^[A-Z][a-zA-Z0-9]*$'.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00101)

File: java-squid/src/main/java/org/sonar/java/model/declaration/VariableTreeImpl.java [hunk 4]
```diff
@@ -47,6 +49,8 @@
   private IdentifierTree simpleName;
   @Nullable
   private ExpressionTree initializer;
+  @Nullable
+  private SyntaxToken endToken;
 
   // FIXME(Godin): never should be null, i.e. should have default value
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/VariableTreeImpl.java]: MAJORTake the required action to fix the issue indicated by this comment.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: java-squid/src/main/java/org/sonar/java/model/expression/MethodInvocationTreeImpl.java [hunk 5]
```diff
@@ -44,12 +44,17 @@
   @Nullable
   private TypeArguments typeArguments;
   private Symbol symbol = Symbols.unknownSymbol;
+  private SyntaxToken openParenToken;
+  private SyntaxToken closeParenToken;
 
-  public MethodInvocationTreeImpl(ExpressionTree methodSelect, @Nullable TypeArguments typeArguments, List<ExpressionTree> arguments, AstNode... children) {
+  public MethodInvocationTreeImpl(ExpressionTree methodSelect, SyntaxToken openParenToken, SyntaxToken closeParenToken, @Nullable TypeArguments typeArguments,
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/expression/MethodInvocationTreeImpl.java]: MAJORorg/sonar/java/model/expression/MethodInvocationTreeImpl must not use com/sonar/sslr/api/AstNode[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 new issues:
* 3 major

Details:
* [The Cyclomatic Complexity of this method "getFirstToken" is 21 which is greater than 10 authorized.](https://github.com/SonarSource/sonar-java/blob/200c3552fcf50313fbfb2bdffb95625ac24da51f/java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java#L157)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* [Reduce the number of returns of this method 11, down to the maximum allowed 5.](https://github.com/SonarSource/sonar-java/blob/200c3552fcf50313fbfb2bdffb95625ac24da51f/java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java#L157)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1142)
* [Should org.sonar.java.checks.TooManyStatementsPerLine_S00122_Check$StatementVisitor be a _static_ inner class?](https://github.com/SonarSource/sonar-java/blob/200c3552fcf50313fbfb2bdffb95625ac24da51f/java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java#L112)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASIC_INNER_SHOULD_BE_STATIC)

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by benzonico** [comment]
LGTM except for comment.

**Pull request closed without merging**